test/Npgsql.Tests/BugTests.cs

```diff
@@ -361,6 +361,45 @@ public void Bug2274()
             }
         }
 
+        [Test]
+        public void Bug2278()
+        {
+            using (var conn = OpenConnection())
+            {
+                try
+                {
+                    conn.ExecuteNonQuery("CREATE TYPE enum_type AS ENUM ('left', 'right')");
+                    conn.ExecuteNonQuery("CREATE DOMAIN enum_domain AS enum_type NOT NULL");
+                    conn.ExecuteNonQuery("CREATE TYPE composite_type AS (value enum_domain)");
+                    conn.ExecuteNonQuery("CREATE TEMP TABLE data (value composite_type)");
+                    conn.ExecuteNonQuery("INSERT INTO data (value) VALUES (ROW('left'))");
+
+                    conn.ReloadTypes();
+                    conn.TypeMapper.MapComposite<Bug2278CompositeType>("composite_type");
+                    conn.TypeMapper.MapEnum<Bug2278EnumType>("enum_type");
+
+                    var xz = conn.ExecuteScalar("SELECT * FROM data AS d");
+
+                }
+                finally
```

Comment:
Wouldn't this suppress any test failure?

Reply:
It wouldn't since it's a `finally` block, not a `catch`.

Removed the `xz` variable and the empty line after it.